File: strings/1768-merge-strings-alternately/merge-strings.go
package main

import "strings"

func main() {

}

// func mergeAlternately1(word1 string, word2 string) string {
//     res := ""
//     max := len(word1)
//     if (len(word2) > max) {
//         max = len(word2)
//     }
//     for i := 0; i<max; i++ {
//         if (i < len(word1)) {
//             res += string(word1[i])
//         }
//         if (i < len(word2)) {
//             res += string(word2[i])
//         }
//     }
//     return res
// }

func mergeAlternately(word1 string, word2 string) string {
	sb := strings.Builder{}
	max := len(word1)
	if len(word2) > max {
		max = len(word2)
	}
	for i := 0; i < max; i++ {
		if i < len(word1) {
			sb.WriteByte(word1[i])
		}
		if i < len(word2) {
			sb.WriteByte(word2[i])
		}
	}
	return sb.String()
}

File: strings/1768-merge-strings-alternately/merge-strings.php
<?php

class Solution {

    /**
     * @param String $word1
     * @param String $word2
     * @return String
     */
    function mergeAlternately($word1, $word2) {
        $w1 = str_split($word1);

        $res = '';
        $len1 = strlen($word1);
        $len2 = strlen($word2);
        $maxLen = max($len1, $len2);
        for($i = 0; $i<$maxLen; $i++) {
            // Если индекс пока в рамках длины первого слова, добавляем букву из него
            if ($i<$len1) {
                $res .= $word1[$i];
            }
            if ($i<$len2) {
                $res .= $word2[$i];
            }
        }
        return $res;
    }
}
